### PR TITLE
update to 1.62.2

### PR DESCRIPTION
**Pull request by cbouss**
grpcio 1.62.2 

**Destination channel:** main

### Links

- https://anaconda.atlassian.net/browse/PKG-4098
- https://github.com/grpc/grpc/tree/v1.62.2
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/abseil-cpp-feedstock/pull/9
  - https://github.com/AnacondaRecipes/libprotobuf-feedstock/pull/24
  - https://github.com/AnacondaRecipes/protobuf-feedstock/pull/21
  - https://github.com/AnacondaRecipes/openssl-feedstock/pull/29

### Explanation of changes:
- Updating to 1.62.2 instead of latest because of https://github.com/protocolbuffers/protobuf/issues/12927
- sync from conda-forge
- adapt recipe

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -109,53 +121,57 @@ outputs:
...
-        - {{ pin_subpackage('grpc-cpp', exact=True) }}
         - python
-        - cython 0.29
         - pip
-        - pthread-stubs 0.3  # [linux]
         - setuptools
-        - wheel >=0.29
-        - abseil-cpp 20230802.0
-        - c-ares 1.19.1
-        - libprotobuf {{ libprotobuf }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:145]: Is `protobuf` required for building `grpcio`? Or it's only for examples and/or testing?
See https://github.com/grpc/grpc/blob/v1.62.2/requirements.txt#L4

**Review by skupr-anaconda — APPROVED** [review]
LGTM but a clarifying question about `protobuf` for `grpcio`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,105 +1,117 @@
...
         # https://discourse.cmake.org/t/questions-about-find-package-cli-msvc/6194
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - cmake
-        - ninja
+        - ninja-base
         - pkg-config
       files:
-        - tests/include/*
+        - cmake_test/
```
Comment (boldorider4) [inline recipe/meta.yaml:98]: I'm failing to see the difference between `files` and `source_files`? Shouldn't `hello.proto` be part of `source_files`?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,105 +1,117 @@
...
-        - m2-patch  # [win]
-        - patch  # [not win]
-        - cmake >=3.5.1
-        - pkg-config  # [unix]
-        # `protoc` is also used for building
-        - libprotobuf {{ libprotobuf }}
-        - ninja
+        - cmake
+        - ninja-base
+        # During cross-compilation, we need to build the grpc_cpp_plugin for the
```
Comment (boldorider4) [inline recipe/meta.yaml:67]: maybe I'm missing something, but when do we do cross compilation?

Comment (boldorider4) [inline recipe/meta.yaml:67]: maybe I'm missing something, but when do we do cross compilation?

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:98]: files takes files from the recipe directory: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-files

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:67]: We don't, but most changes here are synced from conda-forge.

Comment (cbouss) [reply recipe/meta.yaml:67]: We don't, but most changes here are synced from conda-forge.

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:145]: Not from what I see.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Looks good to me. I went though the build logs for linux-64, osx-arm64 and win-64, looked at the DSO analysis, looked at the cython compiler flags and I also took a quick look at the installed files. I also inspected the builds scripts and nothing in particular catched my attention.

I did notice https://github.com/grpc/grpc/blob/v1.62.2/setup.py#L299-L322 on the setup.py though. But I think it should be mostly fine. Same for https://github.com/grpc/grpc/blob/v1.62.2/setup.py#L53-L81.

I don't think we necessarily need to fix this right now. It could well be done in another PR (since this PR is already big enough). I think it's something we could also contribute back in the conda-forge feedstock.

**Review by JamesRobertsonGames — APPROVED** [review]

**Comment by cbouss** [comment]
> Looks good to me. I went though the build logs for linux-64, osx-arm64 and win-64, looked at the DSO analysis, looked at the cython compiler flags and I also took a quick look at the installed files. I also inspected the builds scripts and nothing in particular catched my attention.
> 
> I did notice https://github.com/grpc/grpc/blob/v1.62.2/setup.py#L299-L322 on the setup.py though. But I think it should be mostly fine. Same for https://github.com/grpc/grpc/blob/v1.62.2/setup.py#L53-L81.
> 
> I don't think we necessarily need to fix this right now. It could well be done in another PR (since this PR is already big enough). I think it's something we could also contribute back in the conda-forge feedstock.

Include directories are processed in left to right order, so this works as expected.
https://gcc.gnu.org/onlinedocs/gcc-6.2.0/gcc/Directory-Options.html
https://learn.microsoft.com/en-us/cpp/build/reference/i-additional-include-directories?view=msvc-170
https://clang.llvm.org/docs/ClangCommandLineReference.html#include-path-management

**Pull request merged**